core/templates/dev/head/pages/library/SearchResultsDirective.js

```diff
@@ -20,6 +20,9 @@ oppia.directive('searchResults', [
   'UrlInterpolationService', function(UrlInterpolationService) {
     return {
       restrict: 'E',
+      scope: {
+        learnerDashboardActivityIds: '=learnerDashboardActivityIds',
```

Comment:
Actually, passing this info through all the directives like this is rather ugly.

Here's what I suggest doing. The Library controller should, when the data is retrieved from the backend, initialize a service with this data. The job of that service is to keep track of the learner-dashboard-related stuff, and do nothing else. The service should have methods that allow modification of its internal state. There should be a backendApi service in templates/dev/head/domain that handles updates to the backend.

If sub-directives or components need to refer to the learner dashboard info service, they can do so directly. This way, you can also test that service using Karma tests.

Conceptually, it looks like all the add/remove from learner playlist stuff is a function of just the summary tile. The library controller and the search controller should not need to know about it.